### ISSUE TITLE
neither `bodyTimeout` nor `headersTimeout` works?

### ISSUE THREAD

**Issue by titanism** [nodejs/undici#1987]
## Bug Description

Per discussion in #1971, it is discovered that neither `bodyTimeout` nor `headersTimeout` works to set a max timeout for the request.  Nor does `timeout` option.  Undici should support a max timeout per request from the time of invocation.  Or if not, at least document in the README that signals are required that abort after a certain time.

## Reproducible By

```js
const process = require('node:process');

const undici = require('undici');

const mockAgent = new undici.MockAgent();

mockAgent
  .get('http://test')
  .intercept({ path: '/' })
  .reply(200, 'ok')
  .delay(10000);

undici.setGlobalDispatcher(mockAgent);

setTimeout(function () {
  console.log('bodyTimeout did not work'); // <--- gets called
  process.exit(0);
}, 5000);

undici
  .request('http://test', {
    bodyTimeout: 1000, // <-- none of these work
    headersTimeout: 1000, // <-- none of these work
    timeout: 1000 // <-- none of these work
  })
  .then(() => {
    console.log('request successful, this was not valid test');
    process.exit(0);
  })
  .catch((err) => {
    if (
      err.name === 'ConnectTimeoutError' ||
      err.name === 'HeadersTimeoutError' ||
      err.name === 'BodyTimeoutError' ||
      err.name === ''
    ) {
      console.log('it worked!', err.name); // <---- this never gets invoked
    }

    console.error(err);
    process.exit(0);
  });
```

## Expected Behavior

The output "it worked" should be outputted.

## Logs & Screenshots

```sh
> node test
bodyTimeout did not work
```

## Environment

macOS Node v18

### Additional context

A simpler property such as `timeout` would be nice.

**Comment by ronag** [comment]
How is this a bug? The existing option work as documented? I think what you are looking for is a feature which already exists through signal timeout,

**Comment by ronag** [comment]
Or is this just related to mocking?

**Comment by ronag** [comment]
> Overall the Undici docs are hard to follow, because of all the nesting and inheritence going on, which often takes you in circles if you click the links.

PR welcome :)

**Comment by ronag** [comment]
I think there might be a misunderstand in regards to how body timeout works. It will only timeout if the client is actively reading the body.

**Comment by ronag** [comment]
Also please note that in indicó the response body must always be either consumed or destroyed.

**Comment by kasavetova** [comment]
>  It will only timeout if the client is actively reading the body.

@ronag I have tried body.resume(), body.dump() and body.json() and none of them timeout
